lib/utils.ml

```diff
@@ -21,6 +21,14 @@ struct
     f (rev l1) l2
 
   let (@) = append
+  
+  let concat (xss : 'a list list) : 'a list =
+    let rec tail xss acc =
+      let (@) = List.rev_append in 
+      match xss with
+      | [] -> List.rev acc
+      | xs::xss -> tail xss (xs @ acc)
+    in tail xss []
```

Comment:
this would probably be more clear
```suggestion
  let concat (xss : 'a list list) : 'a list =
    let rec tail xss acc =
      match xss with
      | [] -> List.rev acc
      | xs::xss -> tail xss (List.rev_append xs acc)
    in tail xss []
```

Reply:
still getting a hang of github comments